node-js-peer/src/components/peer-list.tsx

```diff
@@ -0,0 +1,41 @@
+import { useLibp2pContext } from '../context/index.js'
+import { CHAT_TOPIC } from '../constants.js'
+import React, { useEffect, useState } from 'react'
+import { Peer } from './peer.js'
+import type { PeerId } from '@libp2p/interface'
+import { Text, Frame, View } from 'react-curse'
+import { PositionProps } from '../index.js'
+
+export function PeerList(props: PositionProps) {
+  const { libp2p } = useLibp2pContext()
+  const [subscribers, setSubscribers] = useState<PeerId[]>([])
+
+  useEffect(() => {
+    const onSubscriptionChange = () => {
+      setSubscribers(libp2p.services.pubsub.getSubscribers(CHAT_TOPIC))
+    }
+    onSubscriptionChange()
+
+    libp2p.services.pubsub.addEventListener('subscription-change', onSubscriptionChange)
+
+    return () => {
+      libp2p.services.pubsub.removeEventListener('subscription-change', onSubscriptionChange)
+    }
+  }, [libp2p, setSubscribers])
+
+  return (
+    <>
+      <Frame absolute={true} {...props}>
+        <View>
+          <Peer key={libp2p?.peerId?.toString()} peer={libp2p?.peerId} self={true} children={[]} />
+          {subscribers.map((p) => (
+            <Peer key={p.toString()} peer={p} self={false} children={[]} />
```

Comment:
Do you know why the empty array `children` prop is necessary here? Is this a react-curse thing, because I get an error thrown if removed.

Reply:
I think it might be a bug in `react-curse`

Reply:
Let's maybe add a comment about this?

Reply:
Done & have linked to a fix PR'd to react-curse